commons/src/main/java/info/metadude/android/eventfahrplan/commons/SystemClock.kt

```diff
@@ -0,0 +1,112 @@
+package info.metadude.android.eventfahrplan.commons
+
+import org.threeten.bp.*
+import org.threeten.bp.temporal.ChronoField
+
+
+/**
+ * An instance represents a moment in time. All operations are UTC based.
+ * Timezone based dates are retrieved using {#toZonedDateTime}.
+ *
+ * E.g.
+ *
+ * > SystemClock().toZonedDateTime(ZoneOffset.of("GMT+1"))
+ * */
+class SystemClock() : Clock {
```

Comment:
# This class
Did you try to let the class be a `data` class as in `data class SystemClock()`? It automatically generates the `equals()` and `hash()` methods.
Some of the methods where `return =` is used can be converted into expression methods.
Please document all public methods using KDoc. Example input/output is appreciated.

Reply:
- [ ] Did you try to let the class be a data class as in data class SystemClock()? It automatically generates the equals() and hash() methods.
- [x] Some of the methods where return = is used can be converted into expression methods.
- [x] Please document all public methods using KDoc. Example input/output is appreciated. Please note that KDoc uses a prosa form.

Reply:
Nice, thanks.

Can you point out to me where the actual usages of `equals` and `hash()` are? I am think of whether we need to back up these with a separate test.

Reply:
E.g. for a HashMap key. But for now I am not aware of any place that relies on equals / hashCode of Moment class.

Reply:
I am confused. Why did you add the `equals` method in the first place?

Reply:
Because I always do for all my "data" classes (to prevent accidental misbehaviour if its at any time used as e.g. HasMap key).
No we have a kotlin data class, so the equals and hashCode methods are generated by the compiler.

Reply:
Ah. Fine with me. A bit of precaution is fine. Thank you for the explanation.